packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -436,16 +444,32 @@ export class ConnectionManager implements IConnectionManager {
 
 		this._outbound.clear();
 
+		let text = "Closing DeltaManager";
+		let error: ICriticalContainerError | undefined;
+		let finalSwitchToReadonly = switchToReadonly;
+
+		// Handle overloads
+		if (
+			typeof disconnectReasonOrError === "string" &&
+			disconnectReasonOrError in DisconnectReason
+		) {
+			text = disconnectReasonOrError;
+			error = errorOrSwitchToReadonly as ICriticalContainerError;
+		} else {
+			error = disconnectReasonOrError as ICriticalContainerError;
+			finalSwitchToReadonly = (errorOrSwitchToReadonly as boolean) ?? true;
```

Comment:
This doesn't look right. Probably could get away with
`typeof errorOrSwitchToReadonly === "boolean" ? errorOrSwitchToReadonly : true`.
(You should strive to avoid using `as` as much as possible.)